draft-ietf-jose-hpke-encrypt.md

```diff
@@ -345,11 +429,18 @@ HPKE also offers modes that offer authentication.
 HPKE relies on a source of randomness to be available on the device.
 In Key Agreement with Key Wrapping mode, CEK has to be randomly generated and it MUST be ensured that the guidelines in {{RFC8937}} for random number generations are followed.
 
+## HPKE authentication
+
+Authenticated HPKE modes MUST NOT be used with Key Encryption, as this is trivially insecure.
+
 ## Key Management
 
-Reusing a single KEM key across multiple algorithm combinations MUST be avoided to maintain cryptographic security. Each key and its associated algorithm suite, comprising the KEM, KDF, and AEAD, should be managed independently. This separation prevents unintended interactions or vulnerabilities between suites, ensuring the integrity and security guarantees of each algorithm suite are preserved.
-Additionally, the same key MUST NOT be used for both key wrapping and content encryption, as it may introduce security risks.  It
-creates algorithm confusion, increases the potential for key leakage, cross-suite attacks, and improper handling of the key.
+A single key SHOULD NOT be used as both authentication key and recipient key, as this has not been proven to be safe.
+
+A single key SHOULD NOT be used with both JOSE-HPKE and other algorithms as this might enable cross-protocol attacks.
+
+The context binding performed by JOSE-HPKE and HPKE ensures that it is safe to use a single key with multiple JOSE-HPKE algorithms and for both
+Integrated Encryption and Key Encryption.
```

Comment:
Restricting key to one mode does not help against key compromise, as any such compromise must involve mechanisms outside JOSE-HPKE.

Integrated Encryption is more straightforward for one recipient, but can not handle multiple recipients.

Reply:
It is not clear to me why a key needs to be used for both Integrated Encryption and Key Encryption modes ?

It would also introduce additional complexity of multiple key_ops (derivebits and wrapKey).
I don't see any such rule defined in in JWE which allows a key to be re-used for direct key agreement and key encryption modes.

Reply:
It is operationally easier to allow both. And AFAIK there is no good reasons to disallow it (where "it might not be safe", e.g., mixing sender and recipient roles, would count as a good reason).

I think key_ops are just totally busted. And it would not need multiple key_ops: Both invoke HPKE, which I think is derivebits (not sure given the mess that is key_ops).

Reply:
I prefer to keep it simple, managing keys for multiple purposes within the same system increases the complexity of key management and raises the risk of implementation errors.

Reply:
Not tracking the purpose is simpler.

And given this is asymmetric encryption, the purpose would have to be expressible in public keys.

Reply:
I prefer to prioritize simplicity: each key should have a well-defined, single purpose. This approach minimizes risk and aligns with the principle of least privilege in cryptographic systems. If there is a strong operational need to allow key reuse, it should be clearly documented along with the associated trade-offs, potential risks and use of multiple key_ops.

Reply:
AFACIT, there are no trade-offs and no risks beyond those inherit in having an implementation for both.

And what key_ops does HPKE use anyway? I know some implementation used DeriveBits because that was required to enable the needed interfaces.

Reply:
I disagree. Combining multiple key_ops (e.g., encryption and derivation) can lead to unintended consequences in certain scenarios, such as algorithm confusion or key misuse across contexts. I suggest raising this issue on the WG mailing list for further discussion.

Reply:
Neither algorithm confusion nor key misuse across contexts is possible with JOSE-HPKE recipient role keys. To get either, one would have to use the same key with sender role, or with something else, and both are NOT RECOMMENDED.

Reply:
what is the use case to use the same key for JWE Integrated Encryption and Key Encryption ?